From 6120d823c860dbf2616b52f20a172632dba10983 Mon Sep 17 00:00:00 2001
From: Allen Xu <b45815@freescale.com>
Date: Fri, 12 Sep 2014 07:17:38 -0500
Subject: [PATCH] ENGR00331773 qspi: imx6sx: Fixed the issues when compiled
 QSPI as a module

There were two issues when compiled QSPI as a module.

1. two functions in spi_nor.c has not been exported.
2. unregister mtd device count is wrong.

Signed-off-by: Allen Xu <b45815@freescale.com>
(cherry picked from commit f8c6c7204404864a042112cbc06b046cda43a1e4)
---
 drivers/mtd/spi-nor/fsl-quadspi.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index 99e95c8..a7d6f00 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -1,7 +1,7 @@
 /*
  * Freescale QuadSPI driver.
  *
- * Copyright (C) 2013-2014 Freescale Semiconductor, Inc.
+ * Copyright (C) 2013-2015 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -1037,7 +1037,7 @@ static int fsl_qspi_probe(struct platform_device *pdev)
 	return 0;
 
 last_init_failed:
-	for (i = 0; i < q->nor_num; i++)
+	for (i = 0; i < FSL_QSPI_MAX_CHIP; i++)
 		mtd_device_unregister(&q->mtd[i]);
 
 irq_failed:
@@ -1053,7 +1053,7 @@ static int fsl_qspi_remove(struct platform_device *pdev)
 	struct fsl_qspi *q = platform_get_drvdata(pdev);
 	int i;
 
-	for (i = 0; i < q->nor_num; i++)
+	for (i = 0; i < FSL_QSPI_MAX_CHIP; i++)
 		mtd_device_unregister(&q->mtd[i]);
 
 	/* disable the hardware */
-- 
1.8.0

